src/main/java/graphql/execution/FieldCollector.java

```diff
@@ -32,7 +33,7 @@
 
     public MergedSelectionSet collectFields(FieldCollectorParameters parameters, MergedField mergedField) {
         Map<String, MergedField> subFields = new LinkedHashMap<>();
-        List<String> visitedFragments = new ArrayList<>();
+        Set<String> visitedFragments = new HashSet<>();
```

Comment:
We should use `LinkedHashSet` here to be predictable and make debugging easier.

Reply:
done